components/region_cache_memory_engine/src/background.rs

```diff
@@ -397,11 +441,18 @@ impl Runnable for BackgroundRunner {
                                 }
                             }
                         }
-                        core.on_snapshot_loaded(range).unwrap();
+                        core.on_snapshot_load_finished(range);
                     }
                 };
                 self.range_load_remote.spawn(f);
             }
+            BackgroundTask::MemoryCheck => {
+                let mem_usage = self.core.memory_controller.mem_usage();
+                if mem_usage > self.core.memory_controller.soft_limit_threshold() {
+                    // todo: select ranges to evict
+                }
+                self.core.memory_controller.set_memory_checking(false);
```

Comment:
> set_memory_checking

What's the purpose of this call? Cannot find  how `memory_checking` is consumed.

Reply:
`memory_checking` is set by the first one whose fails to acquire the memory due to soft limit and is used to notify there is a memory checking task so no need to schedule another one. Otherwise, every memory acquire after the softlimit will schedule one task so  that the worker will be overwhelmed. Also, we cannot directly check the scheduler if there's running task in the worker as there can be some other types of work such as load task.